picli/configs/base_pipe.py

```diff
@@ -151,6 +141,11 @@ def _build_group_configs(self):
                                           f'\n\nInvalid Key: {e}'
                                 util.sysexit_with_message(message)
                     group_configs.append(run_config)
+        if not len(group_configs):
+            message = f'No group configs found for pi_{self.name} in' \
+                      f'{self.base_config.vars_dir}/group_vars.d/'
+            util.sysexit_with_message(message)
```

Comment:
It appears that this is a sys.exit()... Exiting from within subclasses, methods and functions is less desirable as this introduces side-effects when calling logic. Consider using exceptions and handling those from the CLI... Think about how this lib may be used in the future... do you want it to exit if it where used in a web service for example.

Reply:
Normal exceptions also help in unit testing... otherwise you handle ExitError or whatever which is generic... You might even consider creating a custom exception as well if it warrants it

Reply:
I don't think custom exceptions would be required at this point but I do agree that we should probably look into this in the future. I think this would warrant a separate PR so I'll resolve this conversation in the current PR and create an issue to implement